### PR TITLE
Remove many of the dependency projects

### PR DESCRIPTION
**Pull request by jaredpar**
The dependency projects serve a few purposes:

- Help ensure package unity by serving as a single place for a NuGet reference.
- Make NuGet updates simple by having a single place to change.
- Grouping together related packages to make it easy to take a dependency on say Visual Studio editor.

The first two points are largely obsolete now that our project.json references are strictly validated on every build and have a single update mechanism.  This meant a lot of our depnedency projects were just making Roslyn.sln bigger.

As such I went through and deleted all of them which had only 1 or 2 project.json entries.  I left the VS ones which tend to group 10+ references together.  They are still serving a valid "reference VS" purpose.

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @jasonmalinowski, @tmat, @dotnet/roslyn-infrastructure

**Comment by tmat** [comment]
@davkean

**Comment by jaredpar** [comment]
The closed change is expected.  I will have a supporting change to fix that up.  

@jasonmalinowski any objections?

**Comment by jaredpar** [comment]
Lack of objection is being taken as consent here.  Thanks!

File: src/Compilers/CSharp/Test/CommandLine/project.json [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ﻿{
   "dependencies": {
+    "xunit": "2.1.0",
```
Comment (jasonmalinowski) [inline src/Compilers/CSharp/Test/CommandLine/project.json]: General question: what was the philosophy you chose for whether we should restate things that a dependent project already has? Like I'd guess in this case that some TestUtilities project already had xunit?

**Comment by jasonmalinowski** [comment]
:+1:, but would be interested in what entries in the project.json could be removed just because they're redundant. Obviously the tool will update everything, but if there's less churn it has to do, I'd still think that's a good thing.

File: src/Compilers/CSharp/Test/CommandLine/project.json (hunk 1 above)
Comment (jaredpar) [inline src/Compilers/CSharp/Test/CommandLine/project.json]: @jasonmalinowski for this change I just mapped an reference in csproj to an entry in project.json.  I didn't really give a ton of thought to whether or not we should have essentially redundant references.  I wouldn't be opposed to a policy here and really it should be fairly easy to enforce.

Comment (jasonmalinowski) [inline src/Compilers/CSharp/Test/CommandLine/project.json]: @jaredpar: totally fine by me if that's what you did!

**Pull request merged**